[Bugfix] Work package grouping header

oliverguenther opened the pull request:
Currently, changes to groupBy causes the work package to be immediately
refreshed before the actual groups are collected from the backend.
This watch has been removed, since its unneeded.

The work package table is now tracked by the row id and the current
group, which causes the ng-repeat to re-evaluate when the group changes,
thus fixing the group header bug.

Additionally, we now carry the groupByColumn in the wpTableService,
where the other grouping values are derived as well.

Allow nil aggregation group, which should fix https://community.openproject.com/work_packages/22816/activity

oliverguenther committed:
Refactor how table grouping is applied and refreshed

Currently, changes to groupBy causes the work package to be immediately
refreshed before the actual groups are collected from the backend.
This watch has been removed, since its unneeded.

The work package table is now tracked by the row id and the current
group, which causes the ng-repeat to re-evaluate when the group changes,
thus fixing the group header bug.

Additionally, we now carry the groupByColumn in the wpTableService,
where the other grouping values are derived as well.

oliverguenther committed:
Allow nil aggregation group

This should fix https://community.openproject.com/work_packages/22816/activity

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://oliverguenther-fix-wp-header-bug.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://oliverguenther-fix-wp-header-bug.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

oliverguenther committed:
Hide empty group header when groupName not defined

This avoids showing an empty header for inline create row.
Also forces the create button to disappear when the table is explictly
refreshed (e.g., when grouping on another attribute).
